build.sbt

```diff
@@ -198,11 +210,34 @@ val genyVersion = Def.setting {
   else "0.4.2"
 }
 
+lazy val mtags3 = project
```

Comment:
It would be nice if we can fold this into the `mtags` module with `crossScalaVersions = List(2.12, 2.13, 0.23)`. That would allow us to share more code between scala 2 and 3. We may have to move Scala 2 specific code into a `scala-2` source directory but that's fine.

You can see how we configuire this setup in MUnit here https://github.com/scalameta/munit/blob/f22d9c3d841668f3aebaeaf0819438ca53fab7c5/build.sbt#L125-L146

Reply:
@olafurpg Any idea how to disable the scalafix plugin based on Scala version? That's why I initially moved it into two modules. They do reuse a common directory for any common code, which I think is a bit simpler to do.

Is there actual need to have it in a single module?

Reply:
Also we don't have fiddle with the settings and numerous edge cases if we keep them in separate modules. 

I sharing code currently via:
```
 Compile / unmanagedSourceDirectories += file("mtags-common") / "src/main/scala",
```

Reply:
Do you need to disable the Scalafix plugin? As long as you don't run `scalafix` after `++0.21.0...` it's fine. We currently only run Scalafix on 2.12 (not 2.11 or 2.13). We can update sbt-scalafix to special handle Dotty so that the plugin doesn't do anything by default, it's a small fix.

It's desirable to have a single mtags module:

* no custom logic to resolve mtags3 instead of mtags
* idiomatic code reuse via `mtags/src/main/{scala,scala-2,scala-3,...}` folder structure. I have almost always regretted customizing my sbt build outside of these conventions.

Dotty is in my view just another Scala version and it we should try to use `crossScalaVersions` where possible to support it.

Reply:
My biggest motivation to have a single mtags module is that we already use `crossScalaVersions` to abstract over 2.11/2.12/2.13. It would be a step backwards to add a different abstraction to support 0.x or 3.x

Reply:
> no custom logic to resolve mtags3 instead of mtags

We will still need custom logic to resolve it, but I can see how hard it would to merge them.

Reply:
Merged the directories together, but kept the modules separate, otherwise it will not be possible to have language support in Scala 3 build target, since targets correspond to modules.